deploy/README.md

```diff
@@ -11,6 +11,9 @@
 
   - Make sure `dataset_config/MY_DATASET/deploy.json` is filled out.
 
+    - Remove the comments from deploy.json or you'll run into errors
```

Comment:
Technically JSON spec doesn't allow comments, but we allow them in this repo because it's convenient. (Many other tools also allow JSON comments.) For example, here's a place where [we remove comments](https://github.com/DataBiosphere/data-explorer-indexers/blob/f1fa265b775b4c17f422dde0b484e17ca381f499/indexer_util/indexer_util/indexer_util.py#L35).

Feel free to change `deploy-api.sh` to remove comments. Either way, please revert this change.